File: Fatorial/fatorial.c
/**
 * 
 * autor Maurício Freire
 * Calcula o fatorial de um número.
 * Calculates the fatorial of a number.
*/
#include <stdio.h>

int fatorial(int);

int main()
{
    int f;
    printf("Digite um numero: ");
    scanf("%d", &f);
    printf("%d\n", fatorial(f));
    return 0;
}

int fatorial(int f)
{
    if (f == 1) return 1;
    return f * fatorial(f - 1);
}

File: Fatorial/fatorial.rb
=begin
 *
 * autor Maurício Freire
 * Calcula o fatorial de um número.
 * Calculates the fatorial of a number.
=end

def fatorial(n)
    if n == 1 
        return 1
    end
    return n * fatorial(n - 1)
end

n = gets.chomp.to_i
puts "#{n}! = #{fatorial(n)}"
